metadata/gc.go

```diff
@@ -398,9 +399,26 @@ func remove(ctx context.Context, tx *bolt.Tx, node gc.Node) error {
 		cbkt := nsbkt.Bucket(bucketKeyObjectContent)
 		if cbkt != nil {
 			cbkt = cbkt.Bucket(bucketKeyObjectBlob)
+			if cbkt != nil {
+				// check for temporarily protected content
+				bbkt := cbkt.Bucket([]byte(node.Key))
+				if bbkt != nil {
+					l, err := boltutil.ReadLabels(bbkt)
+					if err != nil {
+						return err
+					}
+					if expV, ok := l[string(labelGCExpire)]; ok {
+						exp, err := time.Parse(time.RFC3339, expV)
+						if err != nil || !time.Now().After(exp) {
+							return err
+						}
+					}
+				}
+			}
```

Comment:
There's quite a few nested things, can we just move this out into it's own code block?

Reply:
Moved it into its own function. Was the trickiest part to figure out how these buckets are all organized...